src/wallet/wallet.cpp

```diff
@@ -4281,11 +4281,11 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(const std::string& name,
         //We can't rescan beyond non-pruned blocks, stop and throw an error
         //this might happen if a user uses an old wallet within a pruned node
         // or if he ran -disablewallet for a longer time, then decided to re-enable
-        if (fPruneMode)
-        {
+        if (fPruneMode) {
             CBlockIndex *block = chainActive.Tip();
-            while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA) && block->pprev->nTx > 0 && pindexRescan != block)
+            while (block->pprev && !IsBlockPruned(block->pprev) && pindexRescan != block) {
```

Comment:
Why remove `block &&` here?

Reply:
Musisz wyreindeksować (ponownie pobrać cały łańcuch blockchain w przypadku oczyszczonego węzła

Reply:
ostatnia synchronizacja portfela wykracza poza dane przycięte

Reply:
@sipa `.Tip()` is always not null, and `block->pprev` is next `block` value.